flink-autoscaler-plugin-jdbc/src/main/java/org/apache/flink/autoscaler/jdbc/event/JdbcAutoScalerEventHandler.java

```diff
@@ -104,6 +135,49 @@ public void handleScalingEvent(
         }
     }
 
+    @Override
+    public void close() {
+        if (Objects.nonNull(scheduledEventHandlerCleaner)
+                && !scheduledEventHandlerCleaner.isShutdown()) {
+            scheduledEventHandlerCleaner.shutdownNow();
+        }
+    }
+
+    @VisibleForTesting
+    void cleanExpiredEvents() {
+        var date =
+                Timestamp.from(
+                        jdbcEventInteractor
+                                .getCurrentInstant()
+                                .minusMillis(eventHandlerTtl.toMillis()));
+        try {
+            Long minId = jdbcEventInteractor.queryMinEventIdByCreateTime(date);
+            if (Objects.isNull(minId)) {
+                log.warn("No expired event handlers queried at {}", new Date());
+                return;
+            }
+            int deleted = 0, batch = 4098;
+            var count = 0L;
+            boolean cleanable = true;
+            for (long startId = minId; deleted == batch || cleanable; startId += batch) {
+                var endId = startId + batch;
+                deleted =
+                        jdbcEventInteractor.deleteExpiredEventsByIdRangeAndDate(
+                                startId, endId, date);
+                count += deleted;
+                if (deleted != batch) {
+                    cleanable = jdbcEventInteractor.queryMinEventIdByCreateTime(date) != null;
+                }
+                log.debug(
+                        "Deleted expired {} event handler records by batch successfully.", deleted);
+                Thread.sleep(10L);
```

Comment:
How about sleep 1s?

One minute could clean 4096 * 60 = 240 k, it's totally enough.